src/MetadataManager/PathToHash.ts

```diff
@@ -0,0 +1,269 @@
+        continue;
+      }
+      if (vscode.workspace.workspaceFolders === undefined) {
+        break;
+      }
+      let path =
+          vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, subMap[2]).path;
+      if (fs.lstatSync(path).isDirectory()) {
+        for (let key in subMap[1]) {
+          queue.push([key, subMap[1][key], subMap[2] + '/' + key]);
+        }
+      } else {
+        flatMap[subMap[2]] = subMap[1];
+      }
+    }
+    return flatMap;
+  }
+
+
+  /**
+   * 
+   * @param uri 
+   * @returns hash :string
+   * @brief put uri in the '_map' to get a hash
+   */
+  get(uri: vscode.Uri) {
+    const splitPath = vscode.workspace.asRelativePath(uri).split('/');
+    let map = this._map;
+
+    splitPath.forEach((path) => {
+      if (map === undefined) {
+        return undefined;
+      }
+      map = map[path];
+    });
+
+    return map;
+  }
+
+  getFilesUnderFolder(uri: vscode.Uri) {
+    const folder = this.get(uri);
+    const files: vscode.Uri[] = [];
+    if (typeof (folder) === 'string') {
+      // not a folder
+      return files;
+    }
+    for (const name in folder) {
+      files.push(vscode.Uri.joinPath(uri, name));
+    }
+
+    return files;
+  }
+
+  /**
+   * @brief Input uri, extract path and hash values, and store them in a '_map'
+   */  
+  async add(uri: vscode.Uri) {
+    const path = vscode.workspace.asRelativePath(uri);
+    const paths = path.split('/');
+    let content: any = await PathToHash.generateHash(uri);
+    let subMap = this._map;
+    let idx = 0;
+    for (let path = paths[idx]; idx < paths.length - 1; path = paths[++idx]) {
+      if (!subMap[path]) {
+        break;
+      }
+      subMap = subMap[path];
+    }
+    if (paths.length - 1 === idx) {  // paths.length - 1: index of a file name
+      // When all of the folder path are stored in pathToHash
+      // update / create pathToHash for a file
+      subMap[paths[idx]] = content;
+      return;
+    }
+
+    for (let i = paths.length - 1; i > idx; --i) {
+      let tempContent: {[key: string]: any} = {};
+      tempContent[paths[i]] = content;
+      content = tempContent;
+    }
+    subMap[paths[idx]] = content;
+}
+
+  /**
+   * @param uri 
+   * @brief Input uri, extract path and hash values, and delete them in a '_map'
+   */
+  delete(uri: vscode.Uri) {
+    let subMap = this._map;
+    const splitPath = vscode.workspace.asRelativePath(uri).split('/');
+
+    for (let i = 0, path = splitPath[i]; i < splitPath.length - 1; path = splitPath[++i]) {
+      if (!subMap) {
+        return;
+      }
+      subMap = subMap[path];
+    }
+    if (subMap === undefined) {
+      // already deleted
+      return;
+    }
+    delete subMap[splitPath[splitPath.length - 1]];
+    if (splitPath.length > 1) {
+      this.deleteEmptyDirPath(this._map, splitPath, 0);
+    }
+  }
+
+  /**
+   * 
+   * @param parent 
+   * @param paths 
+   * @param idx 
+   * @brief delete Empty directory path from '_map'
```

Comment:
Could you write down the comment using only parameters..? `this._map` is onie of an argument pased by the caller. Please specify the function with parameters.